### ISSUE TITLE
artifactVersion in model spec does not redirect to subfolder

### ISSUE THREAD

**Issue by kevinnowland** [SeldonIO/seldon-core#4770]
The `artifactVersion` option in the model spec does not appear to be working as expected.

I deployed a model with the following folder structure to a triton model server:
```
char_count/config.pbtxt
char_count/1/model.py
char_count/2/model.py
```

using a `model.yaml` that looked like

```yaml
apiVersion: mlops.seldon.io/v1alpha1
kind: Model
metadata:
  name: char-count
  namespace: seldon-mesh
spec:
  storageUri: "/var/models/char_count"
  artifactVersion: 1
  requirements:
    - triton
```

with the expectation that `char_count/1/model.py` would be loaded onto the triton server. However, the file `/mnt/agent/models/char-count_1/1/model.py` on the triton server was that of `char_count/2/model.py`.

https://github.com/SeldonIO/seldon-core/blob/ad4fce903fb1d763ba37b7c9d56b7ae67cb7ca82/operator/apis/mlops/v1alpha1/model_types.go#L103

**Comment by cliveseldon** [comment]
Closed as #4774 fixes this.

**Comment by kevinnowland** [comment]
Thank you!
